src/dune_rules/expander.ml

```diff
@@ -14,19 +14,51 @@ module Expanding_what = struct
     | User_action_without_targets of { what : string }
 end
 
+module With_or_without_deps = struct
```

Comment:
This name ends up being awkward. What about just `Deps` with constructors `With`/`Without`?

Reply:
Sounds good, I did that